distributed/core.py

```diff
@@ -663,15 +663,25 @@ async def send_recv(comm, reply=True, serializers=None, deserializers=None, **kw
             response = await comm.read(deserializers=deserializers)
         else:
             response = None
-    except OSError:
+    # Note: this is asyncio.TimeoutError, not the builtin TimeoutError which is a
+    # subclass of OSError
+    except (TimeoutError, OSError):
```

Comment:
Again, defensive programming - I don't have evidence that anything in the ``distributed.comm`` package ever raises ``asyncio.TimeoutError``

Reply:
A send/recv shouldn't (at least for any of our existing comms), but a connection attempt will.

Reply:
Instead of using `distributed.utils.TimeoutError`, perhaps we just use `asyncio.TimeoutError` (which is the same, but namespaced)? Since we're now firmly on asyncio as an eventloop, I'd find it easier to read the namespaced exception, I had to track down the use elsewhere to make sure the right `TimeoutError` was being used.

Reply:
done